Gems/Atom/RPI/Code/Source/RPI.Builders/Pass/PassBuilder.cpp

```diff
@@ -97,27 +97,16 @@ namespace AZ
         // Helper function to get a file reference and create a corresponding job dependency
         bool AddDependency(FindPassReferenceAssetParams& params, AssetBuilderSDK::JobDescriptor* job)
         {
-            AZStd::string_view& file = params.dependencySourceFile;
-            AZ::Data::AssetInfo sourceInfo;
-            AZStd::string watchFolder;
-            bool fileFound = false;
-            AzToolsFramework::AssetSystemRequestBus::BroadcastResult(fileFound, &AzToolsFramework::AssetSystem::AssetSystemRequest::GetSourceInfoBySourcePath, file.data(), sourceInfo, watchFolder);
-
-            if (fileFound)
-            {
-                AssetBuilderSDK::JobDependency jobDependency;
-                jobDependency.m_jobKey = params.jobKey;
-                jobDependency.m_type = AssetBuilderSDK::JobDependencyType::Order;
-                jobDependency.m_sourceFile.m_sourceFileDependencyPath = file;
-                job->m_jobDependencyList.push_back(jobDependency);
-                AZ_TracePrintf(PassBuilderName, "Creating job dependency on file [%s] \n", file.data());
-                return true;
-            }
-            else
-            {
-                AZ_Error(PassBuilderName, false, "Could not find referenced file [%s]", file.data());
-                return false;
-            }
+            // We use an OrderOnce job dependency to ensure that the Asset Processor knows about the
+            // referenced asset, so we can make an AssetId for it later in ProcessJob. OrderOnce is
```

Comment:
Will you need the whole asset ID, or just the UUID of the source asset? I think what you're trying to do makes sense if you just need the UUID of the source asset.

However, if you need the sub ID as well, you could potentially run into problems where an upstream change (probably to the builder itself) results in new sub IDs, and the builder here doesn't properly re-run the jobs because it was OrderOnce.

Reply:
Technically we do need both UUID and SubId but in this case we know that the product we care about uses SubId 0.

Reply:
Gotcha, so if that sub ID shuffles because someone changes the dependency, you'll need to change the code for this builder anyways. It's at the top of my mind because I've spent the last day fixing bugs around image sub ID changing, the primary sub ID for images used to be 0, now it's 1,000, but not everything was updated to handle that change.